ee/clickhouse/views/cohort.py

```diff
@@ -35,6 +48,52 @@ def insert_cohort_people_into_pg(cohort: Cohort):
     cohort.insert_users_list_by_uuid(items=[str(id[0]) for id in ids])
 
 
+def insert_cohort_actors_into_ch(cohort: Cohort, filter_data: Dict):
+    insight_type = filter_data.get("insight")
+
+    if insight_type == INSIGHT_TRENDS:
+        filter = Filter(data=filter_data)
+        entity = get_target_entity(filter)
+        query, params = ClickhouseTrendsActors(cohort.team, entity, filter).actor_query()
+    elif insight_type == INSIGHT_STICKINESS:
+        stickiness_filter = StickinessFilter(data=filter_data, team=cohort.team)
+        entity = get_target_entity(stickiness_filter)
+        query, params = ClickhouseStickinessActors(cohort.team, entity, stickiness_filter).actor_query()
+    elif insight_type == INSIGHT_FUNNELS:
+        funnel_filter = Filter(data=filter_data)
+        if funnel_filter.correlation_person_entity:
+            query, params = FunnelCorrelationActors(filter=funnel_filter, team=cohort.team).actor_query()
+        else:
+            funnel_actor_class = get_funnel_actor_class(funnel_filter)
+            query, params = funnel_actor_class(filter=funnel_filter, team=cohort.team).actor_query()
+    elif insight_type == INSIGHT_PATHS:
+        path_filter = PathFilter(data=filter_data)
+        query, params = ClickhousePathsActors(path_filter, cohort.team, funnel_filter=None).actor_query()
+
+    insert_actors_into_cohort_by_query(cohort, substitute_params(query, params))
```

Comment:
I think it's worth for future clarity to also assert in all cases it's aggregating by people :)

Reply:
sorry what do you mean here?

Reply:
Above we do ` query, params = ClickhousePathsActors(path_filter, cohort.team, funnel_filter=None).actor_query()` - there's no explicit guarantee that this won't return a groups-based query, causing havoc.

My suggestion is to add something like `if is_groups_query: raise ArgumentError()` or something like that.